src/shoobx/immutable/pjpersist.py

```diff
@@ -237,10 +239,22 @@ def __delitem__(self, key):
 
         # need to make sure that all revisions get deleted
         cur = self._pj_jar.getCursor()
-        qry = self._combine_filters(
-            super()._pj_get_resolve_filter(),
-            sb.Field(self._pj_table, self._pj_mapping_key) == key
-        )
+        try:
+            # this is a small pain here
+            # we cannot use `super()._pj_get_resolve_filter()` because that
+            # could miss some important filters
+            # OTOH `self._pj_get_resolve_filter()` adds filtering on
+            # endOnFld == None that prohibits deleting historic versions because
+            # historic versions will have some date in endOn
+            # `_pj_get_resolve_filter` has no parameters,
+            # I don't want to add any, so we opt for `_v_doNotAddEndOnFilter`
+            self._v_doNotAddEndOnFilter = True
+            qry = self._combine_filters(
+                self._pj_get_resolve_filter(),
+                sb.Field(self._pj_table, self._pj_mapping_key) == key
+            )
+        finally:
+            del self._v_doNotAddEndOnFilter
```

Comment:
I think this code should just not exist. You should not store any state while doing deletion or any other query.